src/calendar.cpp

```diff
@@ -257,11 +257,33 @@ bool calendar::is_night() const
     return (seconds > sunset_seconds + TWILIGHT_SECONDS || seconds < sunrise_seconds);
 }
 
+double calendar::current_daylight_level() const
+{
+    double current_daylight_modifier = 1.;
+    
+    switch (season) {
+    case SPRING:
+        current_daylight_modifier = 1.;
+        break;
```

Comment:
I wasn't sure whether to be as explicit, I was just going off the style of the earlier functions, will use yours.

That sunlight() function is called every turn for vehicles with panels from what I can tell, do we really want to interpolate for a 3% change? I don't by any means know what I'm doing, but that sounds performance-heavy for cases where SEASONAL_DAYLIGHT_OFFSET is small.